spec/features/team/invite_a_team_member_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'features/support/not_signed_in'
+require 'features/support/sign_up_helpers'
+require 'support/invite_use_case_spy'
+require 'support/invite_use_case'
+require 'support/notifications_service'
+
+describe "Invite a team member" do
+  context "when logged out" do
+    before do
+      visit new_user_invitation_path
+    end
+
+    it_behaves_like 'not signed in'
+  end
+
+  context "when logged in" do
+    let(:user) { create(:user, :confirmed) }
+    before do
+      sign_in_user user
+      visit new_user_invitation_path
+    end
+
+    it "shows the invites page" do
+      expect(page).to have_content("Send invitation")
+    end
+
+    context "with correct data" do
+      include_examples 'invite use case spy'
+      include_examples 'notifications service'
+      let(:invited_user_email) { "testing@gov.uk" }
+      let(:invited_user) { User.find_by(email: invited_user_email) }
+
+      before do
+        fill_in "Email", with: invited_user_email
+      end
+
+      it "creates an unconfirmed user" do
+        expect {
+          click_on "Send an invitation"
+        }.to change { User.count }.by(1)
+        expect(InviteUseCaseSpy.invite_count).to eq(1)
+        expect(invited_user.confirmed?).to eq(false)
```

Comment:
Similar thoughts to Emile here - I wouldn't test a model directly from an acceptance spec, as it's testing internals rather than the external behaviour of the system.

If I wanted to continue the test further (making sure the user I'd signed up isn't confirmed yet), I'd check their email for the confirmation / try visiting the site as them and make sure I can't access things yet.

Reply:
Yeah I looked at this and thought the same.
Felt ok with this as it is looking at the state of the email that gets sent out.  Feel like that could the definition of done for this functionality (ensuring the email is correct).

Not sure how else you could assert this using the UI

Reply:
Yeah, I'm happy to look at the e-mail - doesn't have to be the UI that we're confirming, can be anything a user has access to.

Happy to explain more about why I think it's best to avoid testing the models in these specs.

Reply:
Agreed on not testing models in feature specs.
You got anything in mind that we can use as an assertion here instead?

Reply:
For this specific line?:
`expect(invited_user.confirmed?).to eq(false)`

That to me says _I want to check this user has to confirm before they can use the site_.

I'd probably break this out into two specs, one that checks what happens when they login, one that checks they get the right email, something like this:

```ruby
context "with correct data" do
  #snip

  it 'sends the invitee a confirmation email' do
    # check this with spies or similar
  end

  context 'then trying to login as the invitee' do
    before do
      login_as "invitee@gov.uk"
      visit homepage
    end

    it 'tells me to confirm my account first' do
      expect(page).to have_content("confirm your account first")
    end
  end
end
```